maven-plugin/src/main/java/org/moditect/mavenplugin/add/AddModuleInfoMojo.java

```diff
@@ -103,6 +106,18 @@
 
     @Override
     public void execute() throws MojoExecutionException, MojoFailureException {
+    	// Check if this plugin should be skipped
+    	if (skip) {
+    		getLog().debug("Mojo 'add-module-info' skipped by configuration");
+    		return;
+    	}
+    	// Don't try to run this plugin, when packaging type is 'pom'
+    	// (may be better to only run it on specific packaging types, like 'jar')
+//    	if (project.getModel().getPackaging().equalsIgnoreCase("pom")) {
```

Comment:
Should this block be uncommented?

Reply:
Ooops, yes it should. I commented it out to better test for my POM packaging test if skip works as intended and forgot to uncomment it again.

Reply:
Ok. Just let me know once you've updated it and then I think it's good to go (unless you want to address that suggested shared base class change, too).

Reply:
I updated the PR.
And no, I did not plan to go for the shared base class at the moment, though I understand the simplification it would bring. But it also depends on the long-term vision how this plugin might evolve - and I am still trying to understand if and how I will use it.
That said, the plugin helped me with my modular Java migration issues. Thank you for your work.